### DISCUSSION TITLE
[OP Stack Deployment] <Deploy the L1 contracts "Invalid data">

### DISCUSSION THREAD

**Discussion by devudilip** [ethereum-optimism/developers#659] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I am following the tutorial to deploy OP stack on Sepolia. 
While deploying L1 contracts I am facing the below issue.

command followed
`forge script scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL --slow`

### Logs

Logs:

> [0] VM::serializeAddress(, ProtocolVersions, ProtocolVersions: [0x99725407DDaA0cBd8AC760a7b8e7D91608B26F32]) 
    │   └─ ← {"AddressManager":"0xc7941C76D4702a19bA48B750762925d637e8cEFa","ProtocolVersions":"0x99725407DDaA0cBd8AC760a7b8e7D91608B26F32","ProtocolVersionsProxy":"0x8Fa4C484f49A7a296c14F75c0CeBB840dDcf7538","ProxyAdmin":"0x271bA4F980A477D4EEc396c7BdeCA213363302c9","SafeProxyFactory":"0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2","SafeSingleton":"0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552","SuperchainConfig":"0x892A62ad7970Bd20CB4E7E807D3E404C31d01E9F","SuperchainConfigProxy":"0xc448b8B46B053d7b311bBBb9EdB3e84022cD9Be8","SystemOwnerSafe":"0x7b7d83cfC71838FEba3bf738DF03BF043b50cBc4"}
    ├─ [0] VM::writeJson(<stringified JSON>, /Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy) 
    │   └─ ← ()
    ├─ [0] console::log(ProtocolVersions deployed at %s, ProtocolVersions: [0x99725407DDaA0cBd8AC760a7b8e7D91608B26F32]) [staticcall]
    │   └─ ← ()
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1CrossDomainMessengerProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1CrossDomainMessengerProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1StandardBridgeProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1StandardBridgeProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L2OutputOracleProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L2OutputOracleProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/OptimismMintableERC20FactoryProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/OptimismMintableERC20FactoryProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/OptimismPortalProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/OptimismPortalProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/SystemConfigProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/SystemConfigProxy.json": No such file or directory (os error 2)"
    ├─ [0] VM::readFile(/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1ERC721BridgeProxy.json) [staticcall]
    │   └─ ← "failed to read from "/Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/L1ERC721BridgeProxy.json": No such file or directory (os error 2)"
    ├─ [0] console::log(Running chain assertions on the ProtocolVersions) [staticcall]
    │   └─ ← ()
    ├─ [312] ProtocolVersions::owner() [staticcall]
    │   └─ ← 0x000000000000000000000000000000000000dEaD
    ├─ [450] ProtocolVersions::required() [staticcall]
    │   └─ ← 0
    ├─ [428] ProtocolVersions::recommended() [staticcall]
    │   └─ ← 0
    ├─ [0] VM::ffi([bash, -c, forge config --json | jq -r .out]) 
    │   └─ ← 0x666f7267652d617274696661637473
    ├─ [0] VM::ffi([bash, -c, echo ProtocolVersions | sed -E 's/[.][0-9]+\.[0-9]+\.[0-9]+//g']) 
    │   └─ ← 0x50726f746f636f6c56657273696f6e73
    ├─ [0] VM::projectRoot() [staticcall]
    │   └─ ← /Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock
    ├─ [0] VM::261a323e(0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000008c2f55736572732f6465767564696c69702f70726f6a656374732f626c6f636b636861696e2f6f705f737461636b2f6f7074696d69736d2f7061636b616765732f636f6e7472616374732d626564726f636b2f666f7267652d6172746966616374732f50726f746f636f6c56657273696f6e732e736f6c2f50726f746f636f6c56657273696f6e732e6a736f6e0000000000000000000000000000000000000000) 
    │   └─ ← 0x0bc445030000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000c496e76616c696420646174610000000000000000000000000000000000000000
    └─ ← "Invalid data"

> == Logs ==
>   Connected to network with chainid 11155111
>   Storing temp deployment data in /Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy
>   DeployConfig: reading file /Users/meopsuer/projects/blockchain/op_stack/optimism/packages/contracts-bedrock/deploy-config/getting-started.json
>   Deploying from Deploy
>   Deployment context: getting-started
>   Deploying a fresh OP Stack including SuperchainConfig
>   Deploying Safe
>   New SystemOwnerSafe deployed at 0x7b7d83cfC71838FEba3bf738DF03BF043b50cBc4
>   Setting up Superchain
>   Deploying AddressManager
>   AddressManager deployed at 0xc7941C76D4702a19bA48B750762925d637e8cEFa
>   Deploying ProxyAdmin
>   ProxyAdmin deployed at 0x271bA4F980A477D4EEc396c7BdeCA213363302c9
>   ProxyAdmin ownership transferred to Safe at: 0x7b7d83cfC71838FEba3bf738DF03BF043b50cBc4
>   Deploying ERC1967 proxy forSuperchainConfigProxy
>      at 0xc448b8B46B053d7b311bBBb9EdB3e84022cD9Be8
>   SuperchainConfig deployed at 0x892A62ad7970Bd20CB4E7E807D3E404C31d01E9F
>   Running chain assertions on the SuperchainConfig
>   Deploying ERC1967 proxy forProtocolVersionsProxy
>      at 0x8Fa4C484f49A7a296c14F75c0CeBB840dDcf7538
>   Deploying ProtocolVersions implementation
>   ProtocolVersions deployed at 0x99725407DDaA0cBd8AC760a7b8e7D91608B26F32
>   Running chain assertions on the ProtocolVersions
> Error: 
> Invalid data

### Additional Information

I have exactly followed the tutorial, and previous steps worked fine

**Comment by caffeinum** [comment]
Did you try this solution? 

> If you see a nondescript error that includes EvmError: Revert and Script failed then you likely need to change the IMPL_SALT environment variable. This variable determines the addresses of various smart contracts that are deployed via CREATE2. If the same IMPL_SALT is used to deploy the same contracts twice, the second deployment will fail. You can generate a new IMPL_SALT by running direnv allow anywhere in the Optimism Monorepo.
> 
> https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#deploy-the-l1-contracts

**Reply by devudilip** [reply]
Yes I did, but the same issue

**Comment by Chomtana** [comment, marked as answer]
The document is obsoleted

Better to follow the new op-deployer guide

https://docs.optimism.io/builders/chain-operators/tools/op-deployer
